stock_storage_type/models/stock_storage_category_allow_new_product.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class StockStorageCategoryAllowNewProduct(models.Model):
+    _name = "stock.storage.category.allow_new_product"
+    _description = "Storage Category Allow New Product Rule"
+    _order = "storage_category_id, sequence"
+
+    def _selection_allow_new_product(self):
+        return self.env["stock.storage.category"]._fields["allow_new_product"].selection
+
+    storage_category_id = fields.Many2one(
+        comodel_name="stock.storage.category",
+        ondelete="cascade",
+        required=True,
+        index=True,
+    )
+    condition_id = fields.Many2one(
```

Comment:
allow to put multiple conditions (like on the location sequence), it's much more practical
```suggestion
    condition_ids = fields.Many2many(
```

Reply:
Done, and added a tooltip on condition fields telling that all conditions have to match to apply the rule.